util/system/info.cpp

```diff
@@ -68,6 +69,48 @@ static inline double CgroupCpus() {
         return 0;
     }
 }
+
+/*
+In cgroups v2 there isn't a dedicated "cpu" directory under /sys/fs/cgroup,
+so the approximation of the number of CPUs may use the cpu.max file.
+The format is the following:
+
+$MAX $PERIOD
+Which indicates that the group may consume up to $MAX in each $PERIOD duration.
+
+The "max" value could be either the string "max" or a number. In the first case
+our approximation doesn't work so we can bail out earlier.
+*/
+static inline double CgroupV2Cpus(const char* cpu_max_path) {
+    try {
+        TVector<TString> cgroup_cpu_max = StringSplitter(TFileInput(cpu_max_path).ReadAll()).Split(' ').Take(2);
+        double max = FromString<int32_t>(StripString(cgroup_cpu_max[0]));
+        double period = FromString<int32_t>(StripString(cgroup_cpu_max[1]));
+
+        if (max <= 0 || period <= 0) {
+            return 0;
+        }
+
+        return max / period;
+    } catch (...) {
+        return 0;
+    }
+}
+
+static inline double CgroupCpus() {
+    auto cgroup2_cpus = CgroupV2Cpus("/sys/fs/cgroup/cpu.max");
+    if (cgroup2_cpus > 0) {
+        return cgroup2_cpus;
+    }
+    auto cgroups1_cpus = CgroupV1Cpus(
+        "/sys/fs/cgroup/cpu/cpu.cfs_quota_us",
+        "/sys/fs/cgroup/cpu/cpu.cfs_period_us"
+    );
```

Comment:
Please put ");" on the previous line, our internal style check for `util` enforces this style.

```
E     --- L
E     +++ R
E     @@ -104,8 +104,7 @@
E          }
E          auto cgroups1_cpus = CgroupV1Cpus(
E              "/sys/fs/cgroup/cpu/cpu.cfs_quota_us",
E     -        "/sys/fs/cgroup/cpu/cpu.cfs_period_us"
E     -    );
E     +        "/sys/fs/cgroup/cpu/cpu.cfs_period_us");
E          if (cgroups1_cpus > 0) {
E              return cgroups1_cpus;
E          }
```

Reply:
Done!